src/Language/PureScript/Publish/Registry/Compat.hs

```diff
@@ -0,0 +1,89 @@
+toBowerPackage PursJson{..} = do
+  bowerName <- Bower.parsePackageName ("purescript-" <> pursJsonName)
+  let
+    bowerDescription = pursJsonDescription
+    bowerMain = []
+    bowerModuleType = []
+    bowerLicense = [ pursJsonLicense ]
+    bowerIgnore = []
+    bowerKeywords = []
+    bowerAuthors = []
+    bowerHomepage = Just pursJsonLocation
+    bowerRepository = Just $ Bower.Repository { repositoryUrl = pursJsonLocation, repositoryType = "git" }
+    bowerDevDependencies = []
+    bowerResolutions = []
+    bowerPrivate = False
+
+  let parseDependencies = traverse (bitraverse (Bower.parsePackageName . ("purescript-" <>)) (pure . Bower.VersionRange))
+  bowerDependencies <- parseDependencies $ Map.toAscList pursJsonDependencies
+  pure $ Bower.PackageMeta {..}
+
+-- | A partial representation of the purs.json manifest format, including only
+-- | the fields required for publishing.
+-- |
+-- | https://github.com/purescript/registry/blob/master/v1/Manifest.dhall
+--
+-- This type is intended for compatibility with the Bower publishing pipeline,
+-- and does not accurately reflect all possible purs.json manifests. However,
+-- supporting purs.json manifests properly introduces breaking changes to the
+-- compiler and to Pursuit.
+data PursJson = PursJson
+  { -- | The name of the package
+    pursJsonName :: Text
+    -- | The SPDX identifier representing the package license
+  , pursJsonLicense :: Text
+    -- | The GitHub repository hosting the package
+  , pursJsonLocation :: Text
+    -- | An optional description of the package
+  , pursJsonDescription :: Maybe Text
+    -- | The registry-compliant version of the package, which is SemVer minus
+    -- | build metadata and prerelease identifiers. This version can be used
+    -- | to associate this manifest with its metadata.
+  , pursJsonVersion :: Text
+    -- | A map of dependencies, where keys are package names and values are
+    -- | dependency ranges of the form '>=X.Y.Z <X.Y.Z'
+  , pursJsonDependencies :: Map Text Text
+  }
+
+asPursJson :: Parse Bower.BowerError PursJson
+asPursJson = do
+  pursJsonName <- key "name" asText
+  pursJsonDescription <- keyMay "description" asText
+  pursJsonLicense <- key "license" asText
+  pursJsonVersion <- key "version" asText
+  pursJsonDependencies <- key "dependencies" (Map.fromAscList <$> eachInObject asText)
+  -- Packages are required to come from GitHub in PureScript 0.14.x, but the
+  -- PureScript registry does not require this, nor does it require that
+  -- packages are Git repositories. This restriction should be lifted when
+  -- we fully support purs.json manifests in the compiler and on Pursuit.
+  --
+  -- For the time being, we only parse manifests that include a GitHub owner
+  -- and repo pair, or which specify a Git URL, which we use to try and get
+  -- the package from GitHub.
+  pursJsonLocation <- key "location" (catchError asOwnerRepo (const asGitUrl))
```

Comment:
When I edited the `purs.json` file, so that the `location.githubRepo` field was deleted, I got this error:
```
There is a problem with your package, which meant that it could not be
published.                                                            
Details:                                                              
  There was a problem with your package manifest file:                
    At the path: ["location"]                                         
    The required key "gitUrl" is missing                              
                                                                      
  Please ensure that your package manifest file is valid.
```
Perhaps instead of `const asGitUrl`, we should report both errors? For example, something like "after failing to find the `location.githubRepo` field, tried to parse location as git url, but failed."

Reply:
Yea, this is a good point. I'll put together a custom error for it.